accepted/00063-hub-iss.md

```diff
@@ -40,35 +40,48 @@ From the requirements:
 
-The current implementation has a code on the ISS Master to "export" files and code on the ISS Slave to "import" them - this concept is not supposed to change.
+A "file importer" component runs on the target Server and, given an exported directory structure, copies files in appropriate locations.
+
+Transfer of the directory structure is described separately in the "Transport" section below.
 
 ### Database data import/export
 
-The ISS Master, given a specified piece of content to sync, should be able to export relevant data from the ISS Master database, to be transported and imported on the ISS Slave side.
+**First core idea**: a "database exporter" component runs on a Server and dumps data in form of a special SQL script. On the target Server, data is then imported by executing the SQL script against the local database.
 
-The current implementation serializes ISS Master database data to XML and deserializes it on the ISS Slave. Relatively complicated logic is in place on both communication ends to convert to and from this format.
+Dumped data will have to be limited to a specified piece of content to sync (eg. a software channel label, or a set of such labels).
 
-**First core idea** of this RFC is to **export database data from the ISS Master in form of a SQL script to be executed as-is on the ISS Slave** (eg. via `psql`).
+**Second core idea**: the generated SQL script will be generated without substantial hardcoding of database schema information int the program. Rather, the exporter will inspecting the schema programmatically and, for example, determine most or all of the involved tables, key relationships, ordering, etc.
 
-**Second core idea is to generate that SQL script from the ISS Master data mainly by inspecting the schema programmatically** (eg. figuring out by schema inspection most of the involved tables, key relationships, ordering, etc.).
+Notes:
+  - the current implementation serializes ISS Master database data to XML and deserializes it on the ISS Slave. Relatively complicated logic is in place on both communication ends to convert to and from this format
+    - intention of the approach in this RFC is to avoid most of this complication
+  - the current implementation also needs an up-to-date mapping of tables and columns in Python code (ORM), and that has to be kept updated
+    - intention of the approach in this RFC is to avoid most of the manual specification of the mapping, reducing maintenance
 
-A proof-of-concept Python script that implements partially those ideas is available at: https://github.com/SUSE/spacewalk/blob/Manager-4.0-iss2/iss2/iss2.py
+A proof-of-concept Python script that implements partially those ideas is available at: https://github.com/moio/mgr-dump
 
 #### First core idea: export as SQL script
 
-The SQL script exported from the ISS Master will have to be idempotent. For example:
+The exported SQL script will have to be idempotent. For example:
```

Comment:
Should be pointed that all file should be wrapped inside a single transaction, using a `Begin;` statement, for example?

Reply:
To me this is an implementation detail, as the transaction could also be handled by the import tool.

Reply:
sounds good